### ISSUE TITLE
diff_morphology mask -> lse?

### ISSUE THREAD

**Issue by xp1632** [xp1632/Aalen_working_log#82]
### Context

**File:** `src/python_modules/param_optimizer/operations/differentiable_morphology.py`  
**Companion NodeSpec (must stay in sync):** `docs/chaldene-reference/NodeSpec/param_optimizer_diff.ts` — the `morphologyCodeGen` helper generates Python code for Chaldene visual programming nodes.

**What the module does:** Implements differentiable morphological operations (dilation, erosion, opening, closing, gradient, top hat, bottom hat) with a single learnable `radius` parameter. The structuring element (SE) is generated analytically via `se[i,j] = sigmoid(5.0 * (radius - dist[i,j]))`, producing soft values in [0, 1]. Input images are [0, 1] float tensors of shape (B, C, H, W).

**Current approach (masking-based):**
```python
# Dilation
masked = unfolded * se_flat + (1 - se_flat) * (-1e2)
out = masked.max(dim=2).values

# Erosion
masked = unfolded * se_flat + (1 - se_flat) * (1e2)
out = masked.min(dim=2).values
```

**Bug history with this approach:**
1. Originally used `±1e9` as the masking constant. Even at active positions where `se_flat ≈ 0.99995`, the residual `(1 - 0.99995) × 1e9 = 50,000` overwhelmed pixel values in [0,1], causing output values of ~44,795 and loss of 2 billion.
2. Reduced to `±1e2` which fixed the explosion, but compound ops (opening, closing) had **zero gradient for radius** because hard `max`/`min` only propagates gradient through the single argmax/argmin element — when the winning element has `se_flat ≈ 1.0`, the `∂(unfolded * se_flat)/∂(se_flat)` contribution is just the pixel value (~0.5), but after two opposing ops (erode→dilate), the SE gradient contributions cancel out.
3. The current `±1e2` version passes all 88 tests, but the gradient flow for `radius` in compound ops is fragile and depends on there being enough spatial variation in the image for argmax positions to differ between erosion and dilation stages.

### Instruction

Replace the masking-based `_soft_dilation` and `_soft_erosion` with a **LogSumExp (LSE)** formulation. This eliminates magic masking constants entirely and provides smooth gradient flow to all SE positions (not just the argmax).

**Mathematical formulation:**

Soft dilation (smooth max weighted by SE):

```
dilation(x) = (1/t) * log( sum_i exp(t * x_i + log(s_i + eps)) )
```

Soft erosion (smooth min = negated smooth max of negated input):

```
erosion(x) = -(1/t) * log( sum_i exp(-t * x_i + log(s_i + eps)) )
```

Where:
- `x_i` = unfolded pixel values at neighborhood position i
- `s_i` = SE value at position i (in [0, 1] from sigmoid)
- `t` = temperature (higher → closer to hard max/min)
- `eps` = small constant (e.g., `1e-8`) for numerical stability in log

**Implementation details:**

```python
def _soft_dilation(input, se, temperature=10.0):
    B, C, H, W = input.shape
    kh, kw = se.shape
    pad_h, pad_w = kh // 2, kw // 2
    padded = F.pad(input, (pad_w, pad_w, pad_h, pad_h), mode='replicate')
    unfolded = F.unfold(padded, kernel_size=(kh, kw)).view(B, C, kh*kw, H*W)
    se_flat = se.flatten().view(1, 1, -1, 1)
    log_se = torch.log(se_flat + 1e-8)
    out = (1.0 / temperature) * torch.logsumexp(temperature * unfolded + log_se, dim=2)
    return out.view(B, C, H, W)

def _soft_erosion(input, se, temperature=10.0):
    B, C, H, W = input.shape
    kh, kw = se.shape
    pad_h, pad_w = kh // 2, kw // 2
    padded = F.pad(input, (pad_w, pad_w, pad_h, pad_h), mode='replicate')
    unfolded = F.unfold(padded, kernel_size=(kh, kw)).view(B, C, kh*kw, H*W)
    se_flat = se.flatten().view(1, 1, -1, 1)
    log_se = torch.log(se_flat + 1e-8)
    out = -(1.0 / temperature) * torch.logsumexp(-temperature * unfolded + log_se, dim=2)
    return out.view(B, C, H, W)
```

**Why `log(se)` instead of masking:** When `se ≈ 0` (inactive), `log(se) → -∞`, which naturally suppresses that position inside logsumexp — no magic constant needed. When `se ≈ 1` (active), `log(se) ≈ 0`, so the pixel value dominates. Gradient flows smoothly through all positions proportional to their softmax weight.

**Temperature choice:** `t = 10.0` is a reasonable starting point for images in [0, 1]. Higher values (20–50) give sharper morphology closer to hard max/min. This should be a fixed constant (like the existing `ALPHA = 5.0`), NOT a learnable parameter.

**What to update:**
1. `differentiable_morphology.py` — replace `_soft_dilation` and `_soft_erosion` with LSE versions. Add `TEMPERATURE = 10.0` as a class constant on `DiffMorphology`. Update the module docstring to describe the LSE formulation.
2. `param_optimizer_diff.ts` — update the `morphologyCodeGen` helper's `_soft_dilation` and `_soft_erosion` Python code strings to match.
3. The `forward()` method, `DiffMorphology` class, SE generation (`_get_se`), test fixtures, and all other code remain unchanged.

**Validation criteria:**
- All 88 existing tests pass (`python -m pytest tests/ -q` from `src/python_modules/param_optimizer/`)
- `diag_test.py` output stays in [0, 1] with loss < 1.0
- `quick_test.py` converges (brightness → ~0.7, gamma → ~1.5)

**Comment by xp1632** [comment]
Later maybe, i don't want to overcomplicate things
